gc_registry/certificate/services.py

```diff
@@ -382,79 +413,34 @@ def process_certificate_action(
         CertificateActionType.RESERVE: reserve_certificates,
     }
 
-    assert (
-        certificate_action.action_type in certificate_action_functions
-    ), "Invalid action type."  # type: ignore
-
-    try:
-        certificate_action_functions[certificate_action.action_type](  # type: ignore
-            certificate_action, write_session, read_session, esdb_client
+    if valid_certificate_action.action_type not in certificate_action_functions.keys():
+        raise ValueError(
+            f"Action type ({valid_certificate_action.action_type}) not in {certificate_action_functions.keys()}"
         )
-    except Exception as e:
-        logger.error(f"Error whilst processing certificate action: {str(e)}")
-        certificate_action.action_response_status = "rejected"
-    else:
-        certificate_action.action_response_status = "accepted"
 
-    db_certificate_action = GranularCertificateAction.create(
-        certificate_action, write_session, read_session, esdb_client
-    )
+    # try:
+    action_function: Callable[..., Any] = certificate_action_functions[
+        valid_certificate_action.action_type
+    ]
+    action_function(certificate_action, write_session, read_session, esdb_client)
+    # except Exception as e:
+    #     logger.error(f"Error whilst processing certificate action: {str(e)}")
 
-    return db_certificate_action[0]  # type: ignore
-
-
-def validate_query(certificate_query: GranularCertificateActionBase) -> bool:
-    # Bunde ID range start must be lower than range end
-    if (
-        certificate_query.source_certificate_bundle_id_range_start
-        and certificate_query.source_certificate_bundle_id_range_end
-    ):
-        if (
-            certificate_query.source_certificate_bundle_id_range_start
-            > certificate_query.source_certificate_bundle_id_range_end
-        ):
-            logger.error("Bundle ID range start must be lower than bundle ID range end")
-            return False
-
-    # Date range start must be lower than or equal to date range end
-    if (
-        certificate_query.certificate_period_start
-        and certificate_query.certificate_period_end
-    ):
-        if (
-            certificate_query.certificate_period_start
-            > certificate_query.certificate_period_end
-        ):
-            logger.error(
-                "Certificate period start must be lower than certificate period end"
-            )
-            return False
-
-    # If using either bundle quantity or percentage, only one can be used
```

Comment:
Understand moving the rest of the validation elements to the data model validators but don't think this one has been captured anywhere

Reply:
Yep - good catch. Will add back in. Soz